File: kora-app-annotation-processor/src/test/java/ru/tinkoff/kora/kora/app/annotation/processor/app/AppWithFactories11.java
package ru.tinkoff.kora.kora.app.annotation.processor.app;

import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;

@KoraApp
public interface AppWithFactories11 {
    @Root
    default Object mock1(GenericClass<String> object) {
        return new Object();
    }

    default <T> GenericClass<T> factory1(java.io.Closeable t) {
        return new GenericClass<>();
    }

    default <T> GenericClass<T> factory2(Long t) {
        return new GenericClass<>();
    }

    class GenericClass<T> {}
}

File: kora-app-symbol-processor/src/test/kotlin/ru/tinkoff/kora/kora/app/ksp/app/AppWithFactories11.kt
package ru.tinkoff.kora.kora.app.ksp.app

import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root
import java.io.Closeable

@KoraApp
interface AppWithFactories11 {
    @Root
    fun mock1(`object`: GenericClass<String>) = Any()

    fun <T> factory1(t: Closeable): GenericClass<T> {
        return GenericClass()
    }

    fun <T> factory2(t: Long): GenericClass<T> {
        return GenericClass()
    }

    open class GenericClass<T>
}
